kore/src/Kore/Step/Simplification/Ceil.hs

```diff
@@ -256,116 +263,187 @@ makeEvaluateBuiltin sideCondition (Domain.BuiltinList l) = Just $ do
     And.simplifyEvaluatedMultiPredicate sideCondition (MultiAnd.make ceils)
 makeEvaluateBuiltin
     sideCondition
-    (Domain.BuiltinSet Domain.InternalAc
-        {builtinAcChild}
-    )
+    (Domain.BuiltinSet internalAc)
   =
     makeEvaluateNormalizedAc
+        mkOpaqueAc
         sideCondition
         (Domain.unwrapAc builtinAcChild)
+  where
+    Domain.InternalAc { builtinAcChild } = internalAc
+    mkOpaqueAc normalizedAc =
+        mkBuiltinSet internalAc
+            { Domain.builtinAcChild = Domain.wrapAc normalizedAc }
 makeEvaluateBuiltin _ (Domain.BuiltinBool _) = Just $ return OrCondition.top
 makeEvaluateBuiltin _ (Domain.BuiltinInt _) = Just $ return OrCondition.top
 makeEvaluateBuiltin _ (Domain.BuiltinString _) = Just $ return OrCondition.top
 
+type MkOpaqueAc normalized variable =
+    Domain.NormalizedAc normalized (TermLike Concrete) (TermLike variable)
+    -> TermLike variable
+
 makeEvaluateNormalizedAc
     :: forall normalized variable simplifier
     .   ( InternalVariable variable
         , MonadSimplify simplifier
         , Traversable (Domain.Value normalized)
         , Domain.AcWrapper normalized
         )
-    =>  SideCondition variable
+    =>  MkOpaqueAc normalized variable
+    ->  SideCondition variable
     ->  Domain.NormalizedAc
             normalized
             (TermLike Concrete)
             (TermLike variable)
-    -> Maybe (simplifier (OrCondition variable))
-makeEvaluateNormalizedAc
-    sideCondition
+    ->  Maybe (simplifier (OrCondition variable))
+makeEvaluateNormalizedAc mkOpaqueAc sideCondition normalizedAc =
```

Comment:
This seems reasonable, but I was wondering if it would be worth moving the map/set code to a new file?